### DISCUSSION TITLE
How to remove all marks of a type on the parent node?

### DISCUSSION THREAD

**Discussion by EasterPeanut** [ueberdosis/tiptap#1776] [General Questions] [answered]
I currently got: 

```js
editor.chain().focus().selectParentNode().unsetCode().toggleHeading({ level: level }).run()
```

but this will reset my cursor position/selection. Is there a way to remove code marks on the parent node while keeping the position/selection untouched? Or is there a better solution to disable inline code in my headings?

**Comment by hanspagel** [comment, marked as answer]
Yes! This is how the CodeBlock disallows marks:

https://github.com/ueberdosis/tiptap/blob/main/packages/extension-code-block/src/code-block.ts#L37

Read more:

https://www.tiptap.dev/api/schema/#marks

**Reply by EasterPeanut** [reply]
Thanks!
